statuslistener/statuslistener.go

```diff
@@ -104,7 +104,7 @@ func (avs *AvailabilityStatusListener) processEvent(statusMessage types.StatusMe
 		return
 	}
 
-	err = (*modelEventDao).FetchAndUpdateBy(&resourceID, updateAttributes)
+	err = (*modelEventDao).FetchAndUpdateBy(resource, updateAttributes)
```

Comment:
So I think we can change this var from a pointer to an interface to just the interface (once again unless we're gonna do it in a followup).

Reply:
(*modelEventDao) -> (modelEventDao) will be in follow up PR.